### ISSUE TITLE
Cancel child build if the parent build fails

### ISSUE THREAD

**Issue by atlesp** [huserben/TfsExtensions#72]
I now end up with "orphaned" build if the parent build fails, and those are blocking other builds from starting (have not that many build agents).

So a task to cancel a child build, that can be configured to run if some of the previous steps fail, would have been nice to have.

**Comment by huserben** [comment]
Hi @atlesp 

Just so that I get the problem correct: Your use case is that you Trigger a Build and don't await the result of it, but in parallel you continue with your "main" build. And if this one fails, you would like to cancel the running one?

**Comment by atlesp** [comment]
hi @huserben
Yes, that is correct. My builds run for a while and are only split to speed up the build, and the result are merged together at the end of the master build.

**Comment by huserben** [comment]
Ok, thanks for the clarification.

I will look into a seperate task that will be cancelling builds that were triggered.
However I'm not sure how long it will take as I don't have too much free time on my hands, so I can not make any promises.

As soon as anything updates I will let you know in here.

**Comment by huserben** [comment]
Hi @atlesp 

I just uploaded a new version of the package that includes a CancelBuild Task.
![grafik](https://user-images.githubusercontent.com/5486874/40306911-c350b99a-5d00-11e8-9015-56106f469594.png)

The configuration is pretty much the same as for the regular triggering of builds, you only need the specifiy the authentication options.
It will read the id's of the builds that were triggered in previous steps, so please make sure to set the flag to store the build id under advanced configuration for all the builds you want to be cancelled:
![grafik](https://user-images.githubusercontent.com/5486874/40306965-fb966fb6-5d00-11e8-87bb-dc7769c05b01.png)

Then in order to achieve your goal, I think you can add this task at the very end and setup the "Control Options" to only run the task when a previous task has failed:
![grafik](https://user-images.githubusercontent.com/5486874/40307005-214ea066-5d01-11e8-9c66-ff2c75911c99.png)

Can you please try it out and let me know whether it worked or not?

**Comment by atlesp** [comment]
Thanks @huserben ! I have added it to our build now and will test it.

**Comment by atlesp** [comment]
A dev accidentally tested it for me now and it worked fine! Thanks again.

**Comment by huserben** [comment]
@atlesp 

Very good.
Thanks for your improvement proposal. Please don't hesitate to open an issue if you have another one, or you find a bug or have a question in general.
